changelog/2025-04-02T09_08_44+02_00_add_ghc967_support

```diff
@@ -0,0 +1 @@
+FIXED: Clash doesn't build on GHC 9.6.7 (but does on 9.6.6) [#2916](https://github.com/clash-lang/clash-compiler/issues/2916)
```

Comment:
```suggestion
FIXED: Clash did not build on GHC 9.6.7 (but did on 9.6.6) [#2916](https://github.com/clash-lang/clash-compiler/issues/2916)
```

This is a suggestion; the shape of "fixed" entries in the Changelog that I prefer is "Clash now does X", where X is the right thing. Describing the situation as it was, I'd pick a past tense. When it's in present tense, there's a risk people read "Clash has been fixed to no longer build on GHC 9.6.7" although that seems like a silly change.

Feel free to ignore this suggestion.

Reply:
I'll squash merge :)